dhall/tests/diff/function.txt

```diff
@@ -2,5 +2,5 @@
 → …
 → …@- 1
     + 0
-  
-→ …
+
+  → …
```

Comment:
I'm not sure yet what's going on here!

Reply:
Could be another bug in `removeTrailingWhitespace` :(

```
λ> Pretty.Text.renderIO stdout $ Pretty.layoutSmart layoutOpts actualDiffDocument 
  …
→ …
→ …@- 1
    + 0
  
→ …
λ> Pretty.Text.renderIO stdout $ Pretty.removeTrailingWhitespace $ Pretty.layoutSmart layoutOpts actualDiffDocument 
  …
→ …
→ …@- 1
    + 0

  → …
```

Reply:
Here's the `SimpleDocStream`:

<details>

```haskell
> const () <$> Pretty.layoutSmart layoutOpts actualDiffDocument 
SText 2 "  " 
    ( SChar '\8230' 
        ( SLine 0 
            ( SAnnPush () 
                ( SChar '\8594' 
                    ( SAnnPop 
                        ( SChar ' ' 
                            ( SChar '\8230' 
                                ( SLine 0 
                                    ( SAnnPush () 
                                        ( SChar '\8594' 
                                            ( SAnnPop 
                                                ( SChar ' ' 
                                                    ( SChar '\8230' 
                                                        ( SChar '@' 
                                                            ( SText 2 "- " 
                                                                ( SAnnPush () 
                                                                    ( SChar '1' 
                                                                        ( SAnnPop 
                                                                            ( SLine 4 
                                                                                ( SText 2 "+ " 
                                                                                    ( SAnnPush () 
                                                                                        ( SChar '0' 
                                                                                            ( SAnnPop 
                                                                                                ( SLine 2 
                                                                                                    ( SLine 0 
                                                                                                        ( SAnnPush () 
                                                                                                            ( SChar '\8594' 
                                                                                                                ( SAnnPop 
                                                                                                                    ( SChar ' ' 
                                                                                                                        ( SChar '\8230' ( SLine 0 SEmpty ) )
                                                                                                                    )
                                                                                                                )
                                                                                                            )
                                                                                                        )
                                                                                                    )
                                                                                                )
                                                                                            )
                                                                                        )
                                                                                    )
                                                                                )
                                                                            )
                                                                        )
                                                                    )
                                                                )
                                                            )
                                                        )
                                                    )
                                                )
                                            )
                                        )
                                    )
                                )
                            )
                        )
                    )
                )
            )
        )
    )
```

Reply:
Here's a screenshot from `meld` comparing the docstreams before and after `removeTrailingWhitespace`:

![image](https://user-images.githubusercontent.com/3664523/68158452-f5a15100-ff4f-11e9-9d17-84e626317bc0.png)

Before, we have `SLine 2 (SLine 0 …`. After, it's `SLine 0 (SLine 2 …`

Reply:
Reported in https://github.com/quchen/prettyprinter/issues/93.